aws-predictions-tensorflow/src/test/java/com/amplifyframework/predictions/tensorflow/asset/InputTokenizerTest.java

```diff
@@ -0,0 +1,114 @@
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test that input conversion to TensorFlow Lite tokens
+ * works as intended.
+ */
+public final class InputTokenizerTest {
+    private static final long LOAD_TIMEOUT_MS = 100;
+
+    private Context mockContext;
+    private AssetManager mockAssets;
+
+    /**
+     * Set up mock behavior before each test.
+     */
+    @Before
+    public void setUp() {
+        // Mock Android context and asset manager
+        mockContext = mock(Context.class);
+        mockAssets = mock(AssetManager.class);
+    }
+
+    /**
+     * Test that text tokenizer converts input text into a 2-D
+     * array of equivalent token values and pads the rest with 0's.
+     * @throws Exception if dictionary fails to load
+     */
+    @Test
+    @SuppressWarnings("MagicNumber") // word tokens
+    public void testInputTextTokenizer() throws Exception {
+        final CountDownLatch loaded = new CountDownLatch(1);
+        final AtomicReference<Map<String, Integer>> tokens = new AtomicReference<>();
+        final AtomicReference<PredictionsException> error = new AtomicReference<>();
+
+        final String inputText = "Where is the bathroom?";
+        final InputStream stream = new FileInputStream("src/test/resources/word-tokens.txt");
+
+        // Make mock context return mock asset manager
+        when(mockContext.getAssets()).thenReturn(mockAssets);
+
+        // Make mock asset manager return pre-determined input stream
+        when(mockAssets.open(anyString())).thenReturn(stream);
+
+        // Load!! (from mock assets)
+        TextClassificationDictionary dictionary = new TextClassificationDictionary(mockContext)
+                .onLoaded(
+                    onLoad -> {
+                        loaded.countDown();
+                        tokens.set(onLoad);
+                    },
+                    onError -> {
+                        loaded.countDown();
+                        error.set(onError);
+                    }
+                );
+        dictionary.load();
+        loaded.await(LOAD_TIMEOUT_MS, TimeUnit.MICROSECONDS);
+        if (error.get() != null) {
+            fail("Failed to load dictionary.");
+        }
+
+        // Assert that load was successful
+        assertEquals(tokens.get(), dictionary.getValue());
+
+        // Tokenize input
+        float[][] input = dictionary.tokenizeInputText(inputText);
+        float[][] expected = new float[1][input[0].length];
+        expected[0][0] = 1; // <START>
+        expected[0][1] = 2; // Where (<UNKNOWN>)
+        expected[0][2] = 9; // is
+        expected[0][3] = 4; // the
+        expected[0][4] = 2; // bathroom (<UNKNOWN>)
+        // Followed by 0's (<PAD>)
```

Comment:
```
int[] values = { 1, 2, 9, 4, 2};
for (int index = 0; index < : values.size(); index++) {
    expected[0][index] = values[index];
}
```

Reply:
I was considering that, but I think the way I currently have it written makes it easier to document what each number means